sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -241,6 +242,9 @@ def handle_result(
               counter.name, counter.combine_fn)
           merged_counter.accumulator.merge([counter.accumulator])
 
+      # Commit partial GBK states
```

Comment:
<!--new_thread; commit:6ce5f121f4998fe51aa9ac13064dce692cd1228b; resolved:0-->
There's an edge case where this won't work: currently, if there is no keyed_existing_state, the constructor in TransformResult creates a new dictionary, which we use for this assignment.

Instead, let's get rid of result.keyed_existing_state and access the state directly with:

```python
keyed_existing_state = self._transform_keyed_states[result.transform]
for k, v in result.partial_keyed_state.iteritems():
  keyed_existing_state[k] = v
```